### PR TITLE
misc: Add release notes for version 23.1

### PR DESCRIPTION
**Pull request by Harshil2107**
- Updated release notes for gem5 version 23.1 with workload resource specialization, suites and new env variables

Change-Id: I9e6097b37432b862b3a212440a79c3c0da836ec9

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# Version 23.1
+## Significant API and user-facing changes
+### `WorkloadResource` added to resource specialization
+- `Workload` and `AbstractWorkload` classes have been deprecated. They have been converted to be wrappers of `obtain_resource` and `WorkloadResource` class respectively.
```
Comment (powerjg) [inline RELEASE-NOTES.md]: How does one fix their code if they are using the old API?

**Comment by Harshil2107** [comment]
Note: Anyone may add a commit to this PR to add their change. Please do not do rebases. This will be submitted as a  Squash merge.

**Comment by giactra** [comment]
Is this just for stdlib? Or for any update to the RELEASE file?

**Comment by Harshil2107** [comment]
I think this is for any updates we want to have in RELEASE-NOTES.md file.

**Comment by abmerop** [comment]
> Note: Anyone may add a commit to this PR to add their change. Please do not do rebases. This will be submitted as a Squash merge.

I'm sure it's possible. But... how? I don't know how to just 'add a commit' I added your repo as a remote and tried to push. Are there notes somewhere? I want to add release notes for gpufs.

**Comment by Harshil2107** [comment]
> I'm sure it's possible. But... how? I don't know how to just 'add a commit' I added your repo as a remote and tried to push. Are there notes somewhere? I want to add release notes for gpufs.

This is the github documentation on pushing to a PR from a fork: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork

I have also selected the option to allow edits by maintainers so others should be able to push to this PR.

I also use the github PR and issue extension on VSCode which allows users to checkout PRs and push to them. The extension id is: GitHub.vscode-pull-request-github

image

You can click the arrow near the description to checkout the PR.

**Comment by BobbyRBruce** [comment]
I personally like using the GitHub command line utility: https://cli.github.com. It allows you to checkout a pull request. From there it's just adding a commit to the checkout branch and `git push` from there.

**Comment by mattsinc** [comment]
@abmerop : just confirming I need to add the other GPU/VIPER Ruby updates besides GPUFS stuff to the release notes?

**Comment by abmerop** [comment]
> @abmerop : just confirming I need to add the other GPU/VIPER Ruby updates besides GPUFS stuff to the release notes?

Yes you will need to do that.  I have only added release notes from my own PRs.

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -34,6 +34,8 @@ Suites is a new category of resource being introduced in gem5. Documentation of
 - Added support for clang 15 and clang 16
 - gem5 no longer supports building on Ubuntu 18.04
 - GCC 7, GCC 9, and clang 6 are no longer supported
+- The stats `board.memory.mem_ctrl.dram.bytesRead` and `board.memory.mem_ctrl.dram.bytesWritten` are changed to `board.memory.mem_ctrl.dram.dramBytesRead` and `board.memory.mem_ctrl.dram.dramBytesWritten` respectively.
+- The stats `board.memory.mem_ctrl.nvm.bytesRead` and `board.memory.mem_ctrl.nvm.bytesWritten` are changed to `board.memory.mem_ctrl.nvm.nvmBytesRead` and `board.memory.mem_ctrl.nvm.nvmBytesWritten` respectively.
```
Comment (powerjg) [inline RELEASE-NOTES.md]: @Harshil2107 

```suggestion
- Two `DRAMInterface` stats have changed names (`bytesRead` and `bytesWritten`). For instance, `board.memory.mem_ctrl.dram.bytesRead` and `board.memory.mem_ctrl.dram.bytesWritten`. These are changed to `dramBytesRead` and `dramBytesWritten` so they don't collide with the stat with the same name in `AbstractMemory`.
- The stats for `NVMInterface` (`bytesRead` and `bytesWritten`) have been change to `nvmBytesRead` and `nvmBytesWritten` as well.
```

**Comment by powerjg** [comment]
Thanks @giactra and @abmerop! @mattsinc please add any updates that you see fit. You can either directly push to this PR or you can put them in a comment and Harshil will copy them in. We're planning to make the release at the end of the day today, if all goes according to plan :)

**Comment by abmerop** [comment]
This looks good from my end. @mattsinc just making sure, any GPU-SE notes to add?  

Otherwise, congrats folks on the release!

**Comment by mattsinc** [comment]
Sorry, I will work on this today.  Trying to release this during final exams has not left me much time to help, sorry.

**Comment by mattsinc** [comment]
I am not 100% sure my updates worked properly when I tried to push them to Harshil's branch, let me know if something is messed up

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,3 +1,130 @@
...
+gem5 Version 23.1 is our first release where the development has been on GitHub.
+During this release, there have been 362 pull requests merged which comprise 416 commits with 51 unique contributors.
+
+## Significant API and user-facing changes
+
+### The gem5 build can is now configured with `kconfig`
+
+Most gem5 builds (e.g., `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows.
+All of the default builds in `build_opts` are unchanged and still available.
+However, if you want to specialize your build, you now have to use `scons <kconfig command>`.
```
Comment (rogerchang23424) [inline RELEASE-NOTES.md]: Sorry for lately reply, I would like to add following description:

> However, if you want to specialize your build, you now have to use `scons <kconfig command>`.

To

> However, if you want to specialize your build. For example, use customized ruby protocol. The command `scons PROTOCOL=<PROTOCAL_NAME> build/ALL/gem5.opt` will not work anymore. you now have to use `scons <kconfig command>` to update the ruby protocol as example. The double dash options (--without-tcmalloc, --with-asan and so on) are still continue to work as normal.

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,3 +1,130 @@
+# Version 23.1
+
+gem5 Version 23.1 is our first release where the development has been on GitHub.
+During this release, there have been 362 pull requests merged which comprise 416 commits with 51 unique contributors.
+
+## Significant API and user-facing changes
+
+### The gem5 build can is now configured with `kconfig`
+
+Most gem5 builds (e.g., `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows.
```
Comment (rogerchang23424) [inline RELEASE-NOTES.md]: Most gem5 builds without customized options (excluding double dash options) (e.g. ...

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,3 +1,130 @@
...
+
+- Most of the instructions in the 1.0 spec implemented
+- Works with both FS and SE mode
+- Compatible with Simple CPUs, the O3, and the minor CPU models
+- User can specify the width of the vector units
+- Future improvements
+  - Widening/narrowing instructions are *not* implemented
+  - The model for executing memory instructions is not very high performance
+  - The statistics are not correct for counting vector instruction execution
+
```
Comment (rogerchang23424) [inline RELEASE-NOTES.md:83]: > Add new RISC-V Zcb instructions
https://github.com/gem5/gem5/pull/399

File: RELEASE-NOTES.md (hunk 4 above)
Comment (Harshil2107) [reply RELEASE-NOTES.md]: Just to confirm do you want the sentence to change to "Most gem5 builds without customized options (excluding double dash options) (e.g. , `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows."?

Comment (rogerchang23424) [reply RELEASE-NOTES.md]: Yes

**Pull request merged**